protocol/error.go

```diff
@@ -15,13 +15,33 @@ const (
 	ErrorMalformedClientMessage
 )
 
+const (
+	Passed ErrorCode = iota + 20
```

Comment:
To align with the error codes in Java implementation. I guess the idea is much like the HTTP status code, +10 should be errors client messages whereas +20 are errors of servers'.